### PR TITLE
[Bluetooth][TCSACR-311] Add SetScanMode API

### PR DESCRIPTION
**Pull request by Wootak**
BluetoothLeAdapter.SetScanMode

https://code.sec.samsung.net/jira/browse/TCSACR-311

### CODE REVIEW

File: src/Tizen.Network.Bluetooth/Tizen.Network.Bluetooth/BluetoothLeAdapterImpl.cs [hunk 1]
```diff
@@ -201,6 +201,18 @@ internal int StopScan()
             return ret;
         }
 
+        internal int SetScanMode(BluetoothLeScanMode mode)
+        {
+            int ret = (int)BluetoothError.None;
+
+            ret = Interop.Bluetooth.SetLeScanMode(mode);
+            if (ret != (int)BluetoothError.None) {
+                Log.Error (Globals.LogTag, "Failed to stop BLE scan - " + (BluetoothError)ret);
```
Comment (dh79pyun) [inline src/Tizen.Network.Bluetooth/Tizen.Network.Bluetooth/BluetoothLeAdapterImpl.cs]: Please modify the log msg

**Comment by Wootak** [comment]
> How about add a overload method of `static public void StartLeScan()` on BluetoothAdapter
> 
> ```cs
>         static public void StartLeScan() 
>         {
>                StartLeScan(BluetoothLeScanMode.Balanced);
>          }
>         static public void StartLeScan(BluetoothLeScanMode mode)
>         {
>             if (BluetoothAdapter.IsBluetoothEnabled && Globals.IsInitialize)
>             {
>                 BluetoothLeImplAdapter.Instance.SetScanMode(mode);
>                 int ret = BluetoothLeImplAdapter.Instance.StartScan ();
>                 if (ret != (int)BluetoothError.None)
>                 {
>                     Log.Error(Globals.LogTag, "Failed to in start the le scan operation, Error - " + (BluetoothError)ret);
>                     BluetoothErrorFactory.ThrowBluetoothException(ret);
>                 }
>             }
>             else
>             {
>                 BluetoothErrorFactory.ThrowBluetoothException((int)BluetoothError.NotEnabled);
>             }
>         }
> ```

Thank

> How about add a overload method of `static public void StartLeScan()` on BluetoothAdapter
> 
> ```cs
>         static public void StartLeScan() 
>         {
>                StartLeScan(BluetoothLeScanMode.Balanced);
>          }
>         static public void StartLeScan(BluetoothLeScanMode mode)
>         {
>             if (BluetoothAdapter.IsBluetoothEnabled && Globals.IsInitialize)
>             {
>                 BluetoothLeImplAdapter.Instance.SetScanMode(mode);
>                 int ret = BluetoothLeImplAdapter.Instance.StartScan ();
>                 if (ret != (int)BluetoothError.None)
>                 {
>                     Log.Error(Globals.LogTag, "Failed to in start the le scan operation, Error - " + (BluetoothError)ret);
>                     BluetoothErrorFactory.ThrowBluetoothException(ret);
>                 }
>             }
>             else
>             {
>                 BluetoothErrorFactory.ThrowBluetoothException((int)BluetoothError.NotEnabled);
>             }
>         }
> ```

Thanks for suggestion!

File: src/Tizen.Network.Bluetooth/Tizen.Network.Bluetooth/BluetoothAdapter.cs [hunk 2]
```diff
@@ -637,6 +637,34 @@ static public void StartLeScan()
             }
         }
 
+        /// <summary>
+        /// Starts the Bluetooth LE scan operation with scan mode.
+        /// </summary>
+        /// <remarks>
+        /// The Bluetooth must be enabled.
+        /// </remarks>The result of the operation StartLeScan.
+        /// <exception cref="NotSupportedException">Thrown when the Bluetooth LE is not supported.</exception>
```
Comment (dh79pyun) [inline src/Tizen.Network.Bluetooth/Tizen.Network.Bluetooth/BluetoothAdapter.cs]: Please add 'param' field

Comment (Wootak) [reply src/Tizen.Network.Bluetooth/Tizen.Network.Bluetooth/BluetoothAdapter.cs]: Done.

**Pull request merged**